proposals/jib_core_library.md

```diff
@@ -0,0 +1,76 @@
+# Proposal: Jib Core as a library for Java
+
+The tracking issue is at [#337](https://github.com/GoogleContainerTools/jib/issues/337).
+
+# Goal
+
+Design for Jib Core as a Java library for building container images.
+
+# Proposed API
+
+`Jib` - the main entrypoint for using Jib Core
+- `JibContainerBuilder from(String baseImageReference)`
+- `JibContainerBuilder from(ImageReference baseImageReference)`
+- `JibContainerBuilder from(RegistryImage baseImage)`
+
+`JibContainerBuilder` - configures the container to build
+- `JibContainerBuilder layer(List<Path> files, Path pathInContainer)`
+- `JibContainerBuilder layer(LayerConfiguration)`
```

Comment:
These two were changed to `addLayer`, right?

Reply:
Yep, I'll change this over.